io.openems.backend.metadata.ldap/src/io/openems/backend/metadata/ldap/LdapUserReader.java

```diff
@@ -0,0 +1,260 @@
+                ldapUsersOuFilter.replace("%%username%%", username),
+                LdapUtils.createSearchControls(
+                    "uid", "memberOf",
+                    "displayName", "givenName", "sn",
+                    "street", "postalCode", "l", "st",
+                    "telephoneNumber", "mail",
+                    "preferredLanguage"
+                )
+            );
+
+            if (!searchResults.hasMore()) {
+                log.warn("User " + username + " not found.");
+                throw OpenemsError.COMMON_AUTHENTICATION_FAILED.exception();
+            }
+
+            SearchResult searchResult = searchResults.next();
+            Attributes resultAttributes = searchResult.getAttributes();
+
+            // id
+            String id = LdapUtils.extractValueFromAttributes(resultAttributes, "uid", username);
+            if (id == null) {
+                throw OpenemsError.COMMON_AUTHENTICATION_FAILED.exception();
+            }
+
+            // name
+            final String name = LdapUtils.extractValueFromAttributes(resultAttributes, "displayName", id, username);
+
+            // firstname
+            final String firstname = LdapUtils.extractValueFromAttributes(resultAttributes, "givenName", "", username, false);
+
+            // lastname
+            final String lastname = LdapUtils.extractValueFromAttributes(resultAttributes, "sn", "", username, false);
+
+            // phone
+            final String phone = LdapUtils.extractValueFromAttributes(resultAttributes, "telephoneNumber", username, false);
+
+            // email
+            final String email = LdapUtils.extractValueFromAttributes(resultAttributes, "mail", username, false);
+
+            // street
+            final String street = LdapUtils.extractValueFromAttributes(resultAttributes, "street", username, false);
+
+            // zip
+            final String zip = LdapUtils.extractValueFromAttributes(resultAttributes, "postalCode", username, false);
+
+            // city
+            final String city = LdapUtils.extractValueFromAttributes(resultAttributes, "l", username, false);
+
+            // country
+            final String country = LdapUtils.extractValueFromAttributes(resultAttributes, "st", username, false);
+
+            // token
+            final String token = UUID.randomUUID().toString();
+
+            // language
+            final String preferredLanguage = LdapUtils.extractValueFromAttributes(resultAttributes, "preferredLanguage", username, false);
+            Language language = Language.from(Optional.ofNullable(preferredLanguage));
+            if (language == null) {
+                language = Language.DEFAULT;
+                log.info("language attribute not found for user " + username + ", using " + language + ".");
+            }
+
+            // globalRole
+            final List<String> memberOf = LdapUtils.extractValuesFromAttributes(resultAttributes, "memberOf", username);
```

Comment:
memberOf is generated and maintained automatically if you install the memberOf overlay. Otherwise you have to create it on your own. I've added a note in the README on this.

Reply:
Thanks, I managed to this and then it works!

Reply:
Thanks, I've managed to do this and it works!